db/go_level_db.go

```diff
@@ -155,10 +154,6 @@ func (db *GoLevelDB) Iterator() Iterator {
 	return &goLevelDBIterator{db.db.NewIterator(nil, nil)}
 }
 
-func (db *GoLevelDB) IteratorPrefix(prefix []byte) Iterator {
```

Comment:
why delete? nobody using it?

Reply:
I'm using this in iavl.

Reply:
I think we should take a look at levigo/goleveldb's iterators and what's supported there... first.

Reply:
LeviGo supports Seek... https://github.com/jmhodges/levigo/blob/master/iterator.go#L121

Reply:
So does Goleveldb: https://godoc.org/github.com/syndtr/goleveldb/leveldb/iterator#IteratorSeeker

Lets adopt these as tmlibs/db.Iterator, and make MemDB support it too...

Reply:
I'll write an exposed module function called "IteratePrefix" that takes an iterator and calls Seek() accordingly to get the same effect.